contracts/Vesting.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/// @title Vesting
+/// @author Eldar Dryga
+/// @notice this contract allows you to distribute tokens to users within a certain time
+
+contract Vesting is Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    IERC20 private immutable _token;
+
+    uint32 public periodOfDistributionInDays;
```

Comment:
As far as I know "tight packing" works only for structs, it do not save slots in storage so The only thing u made - avoid converts from one uint to another
There are some situations when its easy to make `uint32 users` variable cause a lot of math will happen and u will need to convert variables to calculate something, but in general storage variables can be simply uint256